src/main/java/io/github/oliviercailloux/y2018/apartments/utils/JsonConvert.java

```diff
@@ -136,17 +95,19 @@ public static Apartment jsonToApartment(Path jsonPath) throws IOException {
 	public static List<Apartment> jsonToApartments(Path jsonPath) throws IOException {
 		String jsonString = Files.readString(jsonPath);
 		List<Apartment.Builder> apartmentsBuild;
-		List<Apartment> apartments = new ArrayList<Apartment>();
+		List<Apartment> apartments = new ArrayList<>();
 		LOGGER.info("Create ArrayList of Apartment");
 
-		Jsonb jsonb = JsonbBuilder.create();
-		LOGGER.info("Create Json builder");
-
-		apartmentsBuild = jsonb.fromJson(jsonString, new ArrayList<Apartment.Builder>() {
-		}.getClass().getGenericSuperclass());
+		try (Jsonb jsonb = JsonbBuilder.create()) {
+			LOGGER.info("Create Json builder");
+			apartmentsBuild = jsonb.fromJson(jsonString, new ArrayList<Apartment.Builder>() {
+			}.getClass().getGenericSuperclass());
+		} catch (Exception e) {
+			throw new IllegalStateException(e);
```

Comment:
Comment ceci pourrait-il être un problème d’état ? J’imagine plus probable que le path fourni ne contienne pas du json valide, auquel cas c’est un problème d’argument.

Reply:
Fix in commit 46840ca515b51330be5b9f1aed4249af25db0eae